packages/assets-controllers/src/NftDetectionController.ts

```diff
@@ -300,17 +319,44 @@ export class NftDetectionController extends BaseController<
    */
   isMainnet = (): boolean => this.config.chainId === ChainId.mainnet;
 
+  isMainnetByNetworkClientId = (networkClient: NetworkClient): boolean => {
+    return networkClient.configuration.chainId === ChainId.mainnet;
+  };
+
+  private getCorrectChainId({
+    chainId,
+    networkClientId,
+  }: {
+    chainId?: Hex;
+    networkClientId?: NetworkClientId;
+  }) {
+    if (networkClientId) {
+      return this.getNetworkClientById(networkClientId).configuration.chainId;
+    } else if (chainId) {
+      return chainId;
+    }
+    return this.config.chainId;
+  }
+
   /**
    * Triggers asset ERC721 token auto detection on mainnet. Any newly detected NFTs are
    * added.
+   *
+   * @param networkClientId - The network client ID to detect NFTs on.
+   * @param _address - The address to detect NFTs for.
    */
-  async detectNfts() {
+  async detectNfts(networkClientId?: NetworkClientId, _address?: string) {
```

Comment:
why the underscore for `_address`? maybe `accountAddress` just to clarify its not a contract address

Reply:
yea `accountAddress` would work.

this function defines `address` lower in the function as a destructured object on line 373 which is the contract address of the nft.